Sources/Classes/RSContext.m

```diff
@@ -192,7 +246,10 @@ - (nonnull id)copyWithZone:(nullable NSZone *)zone {
     copy.device = self.device;
     copy.network = self.network;
     copy.timezone = self.timezone;
```

Comment:
We should use `synchronized` here for the whole method.

Reply:
I tried doing as you suggested. We're getting multiple race conditions and error if we just synchronise the block.
Actually line no: 250 & 251 needs to be properly synchronised as `traits` and `externalIds` is being accessed and updated from multiple places.

Reply:
Okay.